src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -220,6 +227,89 @@ private ExtensionReference[] ParseExtensions(string metadataFilePath)
             }
         }
 
+        private void ValidateBundleRequirements(ExtensionBundleDetails bundleDetails)
+        {
+            if (_extensionRequirements.BundleRequirementsByBundleId.ContainsKey(bundleDetails.Id))
+            {
+                BundleRequirement requirement = _extensionRequirements.BundleRequirementsByBundleId[bundleDetails.Id];
+                var bundleVersion = new Version(bundleDetails.Version);
+                var minimumVersion = new Version(requirement.MinimumVersion);
+
+                if (bundleVersion < minimumVersion)
+                {
+                    _logger.MinimumBundleVersionNotSatisfied(bundleDetails.Id, bundleDetails.Version, requirement.MinimumVersion);
+                    throw new HostInitializationException($"Referenced bundle {bundleDetails.Id} of version {bundleDetails.Version} does not meet the required minimum version of {requirement.MinimumVersion}. Update your extension bundle reference in host.json to reference {requirement.MinimumVersion} or later. For more information see https://aka.ms/func-min-bundle-versions.");
+                }
+            }
+        }
+
+        private void ValidateExtensionRequirements(List<Type> startupTypes)
+        {
+            var errors = new List<string>();
+
+            void CollectError(Type extensionType, Version minimumVersion, ExtensionStartupTypeRequirement requirement)
+            {
+                _logger.MinimumExtensionVersionNotSatisfied(extensionType.Name, extensionType.Assembly.FullName, minimumVersion, requirement.PackageName, requirement.MinimumPackageVersion);
+                string requirementNotMetError = $"ExtensionStartupType {extensionType.Name} from assembly '{extensionType.Assembly.FullName}' does not meet the required minimum version of {minimumVersion}. Update your NuGet package reference for {requirement.PackageName} to {requirement.MinimumPackageVersion} or later.";
+                errors.Add(requirementNotMetError);
+            }
+
+            foreach (var extensionType in startupTypes)
+            {
+                if (_extensionRequirements.ExtensionRequirementsByStartupType.ContainsKey(extensionType.Name))
+                {
+                    ExtensionStartupTypeRequirement requirement = _extensionRequirements.ExtensionRequirementsByStartupType[extensionType.Name];
+                    Version minimumAssemblyVersion = new Version(requirement.MinimumAssemblyVersion);
+
+                    Version extensionAssemblyVersion = extensionType.Assembly.GetName().Version;
+                    string extensionAssemblySimpleName = extensionType.Assembly.GetName().Name;
```

Comment:
Call `GetName()` once and reuse the `AssemblyName` result to get `Version` and `Name`